File: prms-web/src/app/pages/management/app-params/services/app-param.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '../../../../core/services/base-service';
import { AppParamWritePayload } from '../models/app-param.model';
import { AppParamSearchRequest } from '../models/app-param-search.request';

@Injectable({
  providedIn: 'root',
})
export class AppParamService {
  private readonly apiUrl = '/api/v1/app-params';

  constructor(private base: BaseService) {}

  getPage(request: AppParamSearchRequest) {
    return this.base.post(this.apiUrl + '/page', request);
  }

  getById(id: string) {
    return this.base.get(`${this.apiUrl}/${id}`);
  }

  create(request: AppParamWritePayload) {
    return this.base.post(this.apiUrl, request);
  }

  update(id: string, request: AppParamWritePayload) {
    return this.base.put(`${this.apiUrl}/${id}`, request);
  }

  delete(id: string) {
    return this.base.delete(`${this.apiUrl}/${id}`);
  }
}

File: src/main/java/com/tranhuudat/prms/service/AppParamService.java
package com.tranhuudat.prms.service;

import com.tranhuudat.prms.dto.BaseResponse;
import com.tranhuudat.prms.dto.app_param.AppParamDto;
import com.tranhuudat.prms.dto.app_param.AppParamSearchRequest;

import java.util.UUID;

public interface AppParamService {
    BaseResponse create(AppParamDto request);
    BaseResponse update(UUID id, AppParamDto request);
    BaseResponse delete(UUID id);
    BaseResponse getById(UUID id);
    BaseResponse getPage(AppParamSearchRequest request);
}

